guides/common/modules/proc_enabling-rh-cloud-and-insights-client-reports-on-hosts.adoc

```diff
@@ -0,0 +1,20 @@
+[id="enabling-rh-cloud-and-insights-client-reports-on-hosts_{context}"]
+= Enabling RH Cloud and Insights client reports on hosts
+
+You can enable the Red Hat Insights client on hosts and have the hosts send RH Cloud reports to the {RHCloud}.
+
+ifdef::katello,foreman-el,foreman-deb[]
+Red Hat Insights is a service by Red Hat for {RHEL} hosts.
+Ensure to set this parameter for {RHEL} hosts only.
+If you set the parameter on any non-{RHEL} operating systems, it automatically uploads new reports to the {RHCloud}.
```

Comment:
```suggestion
If you set the parameter on any non-{RHEL} operating systems, {Project} automatically uploads new reports to the {RHCloud}.
```
This line seems confusing to me.

Does it mean that Foreman will be able to install Insights client on Debian/Ubuntu for example? If not, are the reports even generated? Is the host still included in the inventory upload? If not, then what is there to upload?
cc @chris1984 @jeremylenz Can you please help answer these questions? ^^

Also, it depends on the Inventory Upload setting whether Foreman uploads the inventory data automatically or not. It isn't hard coded.
![image](https://github.com/user-attachments/assets/1e623de7-5781-478b-9e2a-ea7f25a23bb9)

Reply:
The Insights client will not be installed on Debian / Ubuntu since the corresponding snippet has a verification if the host is running RHEL.

For the inventory upload, however, Foreman RH-Cloud searches for hosts with the parameter `host_registration_insights != false` independent of the fact when the parameter was actually set or if the insights client is indeed installed.

The facts it uploads to RH Hybrid Cloud Console like the (potentially obfuscated) IP address, hostname seem to be taken from the Foreman database since in our tests they appear in the inventory in RH HCC even for Debian or Ubuntu hosts.

The query loading the host information is in `lib/foreman_inventory_upload/generators/queries.rb` inside the rh-cloud plugin if somebody wants to have a closer look.

As far as I understand `{Project}` uploads an insights report if one is available and a base set of facts whereof IP and hostname can be obfuscated and packages can be excluded. The latter is not relevant for Debian or Ubuntu since they have `deb_packages` instead of `packages`.

I hope that helps in finding a wording for that line and answering your questions.

Reply:
Applied Lena's suggestion.